### ISSUE TITLE
Source file - Multiline doesn't work properly after a while

### ISSUE THREAD

**Issue by Yohann-44** [vectordotdev/vector#8315]
### Vector Version

```
vector 0.14.0 
```

### Vector Configuration File

```toml
[sources.fileLog]
	type         = "file"
	include      = ["/path/logs/file.log"]
	ignore_older = 86400
	read_from = "beginning"
	fingerprinting.strategy = "device_and_inode"
	
	multiline.timeout_ms = 120000
	multiline.start_pattern = "\\) called"
	multiline.mode = "halt_before"
	multiline.condition_pattern = "\\) called"

[sinks.print]
	type = "console"
	inputs = ["fileLog"]
	encoding.codec="json"
```

### Debug Output

```
RUST_BACKTRACE=full vector -vvv 
https://gist.github.com/Yohann-44/e11ed507fbf341c588c39c3ce7ffb6e3
```

### Expected Behavior

All message should started with the pattern ") called" and finish before the next.
The text between the terminals should form a single log.

### Actual Behavior

Everything works at the beginning. But at one point, the multiline loses its mind and I get every line of the file (not the multiline).

I increased the multiline.timeout_ms to 120000. the number of errors reduced but the problem persists after a while. Maybe the time that the timeout_ms is reached.

### Example Data

I wrote a script bash to simulate the real log.
```
#!/bin/bash

let max_iter=500
let i=0
while test $i -lt $max_iter
do
echo    "Starting point ) called" >> ./logs/file.log
        let j=0
        let max_iter_internal=50
        while test $j -lt $max_iter_internal
        do
                echo "useless_text" >> ./logs/file.log
                let j=j+1
        done
    let i=i+1
done
```

**Comment by jszwedko** [comment]
Hi @Yohann-44 ! I _think_ this might be fixed in `master` by #7944 . Do you mind trying the latest nightly build to see if it works for you? https://packages.timber.io/vector/nightly/latest

**Comment by jszwedko** [comment]
I'll close this out since it was verified on #7944. It should be released this week.
